packages/cysync-core/src/dialogs/WalletConnect/context/index.tsx

```diff
@@ -0,0 +1,240 @@
+    accounts: IAccount[];
+  }[];
+  evmAccountsGroup: {
+    assetId: EvmId;
+    accounts: IAccount[];
+  }[];
+  setSelectedEvmAccounts: React.Dispatch<React.SetStateAction<IAccount[]>>;
+  handleSelectAccount: (id: string) => void;
+  handleDisselectAccount: (id: string) => void;
+  getBalanceToDisplay: (account: IAccount) => string;
+  evmAccountDropdownListGroup: {
+    assetId: EvmId;
+    accounts: DropDownListItemProps[];
+  }[];
+  dapp: {
+    logo: string;
+    url: string;
+    name: string;
+  };
+  setDapp: React.Dispatch<
+    React.SetStateAction<{
+      logo: string;
+      url: string;
+      name: string;
+    }>
+  >;
+  supportedNoAccountBlockchain: string[];
+  setSupportedNoAccountBlockchain: React.Dispatch<
+    React.SetStateAction<string[]>
+  >;
+}
+
+export const WalletConnectDialogContext: Context<WalletConnectDialogContextInterface> =
+  createContext<WalletConnectDialogContextInterface>(
+    {} as WalletConnectDialogContextInterface,
+  );
+
+export interface WalletConnectDialogProviderProps {
+  children: ReactNode;
+}
+
+export const WalletConnectDialogProvider: FC<
+  WalletConnectDialogProviderProps
+> = ({ children }) => {
+  const lang = useAppSelector(selectLanguage);
+  const dispatch = useAppDispatch();
+  const [dapp, setDapp] = useState<{
+    logo: string;
+    url: string;
+    name: string;
+  }>({
+    logo: UniSwapLogo,
+    url: 'app.uniswap.org',
+    name: 'Uniswap',
+  });
+  const [supportedNoAccountBlockchain, setSupportedNoAccountBlockchain] =
+    useState<string[]>(['Avalanche C-Chain', 'Solana', 'Binance']);
+
+  const {
+    selectedWallet,
+    setSelectedWallet,
+    handleWalletChange,
+    walletDropdownList,
+    selectedEvmAccounts,
+    selectedEvmAccountsGroup,
+    setSelectedEvmAccounts,
+    getBalanceToDisplay,
+    handleSelectAccount,
+    handleDisselectAccount,
+    evmAccountDropdownListGroup,
+    evmAccountsGroup,
+  } = useEvmAccountDropdown();
+
+  /** @doubt What is it about? */
+  const deviceRequiredDialogsMap: Record<number, number[] | undefined> = {
+    // 1: [0],
+    // 2: [0],
+  };
+
+  const onClose = () => {
+    dispatch(closeDialog('walletConnect'));
+  };
+
+  const tabs: ITabs = [
+    {
+      name: lang.strings.walletConnect.uriTab.title,
+      dialogs: [<WalletConnectPasteURIDialog key="uriTab" />],
+    },
+    {
+      name: lang.strings.walletConnect.accountSelectionTab.title,
+      dialogs: [
+        <WalletConnectAccountSelectionDialog key="accountSelectionTab" />,
+      ],
+    },
+    {
+      name: lang.strings.walletConnect.accountConnectedTab.title,
+      dialogs: [
+        <WalletConnectAccountConnectedDialog key="accountConnectedTab" />,
+      ],
+    },
+  ];
+
+  const {
+    onNext,
+    onPrevious,
+    goTo,
+    currentTab,
+    currentDialog,
+    isDeviceRequired,
+  } = useTabsAndDialogs({
+    deviceRequiredDialogsMap,
+    tabs,
+  });
+
+  const [walletConnectURI, setWalletConnectedURI] = useState<string>();
```

Comment:
To solve the error 
![image](https://github.com/Cypherock/cypherock-cysync/assets/80461749/f0124c57-d337-4c71-8ee7-2f84842f541d)


```suggestion
  const [walletConnectURI, setWalletConnectedURI] = useState<string>('');
```

Reply:
fixed